### ISSUE TITLE
SimplifyMemberName: behaves incorrect on VB object initializer

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#23659]
IDE0037 Visual Basic AND  VB.NET Member name can be simplified

Example: 
 Dim cmd As New SqlCommand With {
                .Connection = conn,
                .Transaction = trans,
                .CommandTimeout = CommandTimeOut
            }

Reports: Member can be simplified

Suggests: 
 Dim cmd As New SqlCommand With {
                .Connection = conn,
                .Transaction = trans,
                CommandTimeOut
            }

Build Fails

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/160818/vbnet-intellisence-reports-and-suggests-wrong.html
VSTS ticketId: 533435_
_These are the original issue comments:_

Shane Hancock on 12/5/2017, 05:40 PM (2 days ago): Same here
_These are the original issue solutions:_
(no solutions)

**Comment by j-dc** [comment]
Dirty fix   (hiding the warnings):    
Project properties -> code analysis   
"Open" rule set
Microsoft.CodeAnalysis.VisualBasic.Features  -> uncheck  IDE0037

**Comment by reduckted** [comment]
Caused by #20649.

**Comment by reduckted** [comment]
To clarify, this is broken in VB for known types, because known types cannot use this shorthand syntax. 

This code produces the suggestion, but after making the suggested change, the code fails to compile:

```vb
Public Class Foo
    Public Property Bar As Integer
End Class

Dim bar As Integer

' Before
Dim f = New Foo With {
    .Bar = bar ' <-- Suggestion: Member name can be simplified.
}

' After
Dim f = New Foo With {
    bar ' <-- Error: Name of field or property being initialized in an object initializer must start with '.'.
}
```

The suggested changes work OK for anonymous types.

One other issue that I have with this analyzer is that it can result in the case of the property name changing. In my opinion, this is wrong. 

Given this code:

```vb
Dim f = New With {
    .Bar = bar ' <-- Suggestion: Member name can be simplified.
}
```

It starts out with an anonymous object that looks like this:

```vb
Class AnonymousType
    Property Bar As Integer
End Class
```

But after making the suggested change, you end up with a type like this:

```vb
Class AnonymousType
    Property bar As Integer
End Class
```

If you're serializing this object, or doing something that involves looking for specific property names, then the object is now different, and that would probably cause problems. In my opinion, this suggestion should only be made if the property name and variable name are considered equal using a case-sensitive comparison (like the C# one does).

**Comment by jcouv** [comment]
Fix is pending review. https://github.com/dotnet/roslyn/pull/23817

The UseInferredMemberName suggestion should only come up for anonymous type creation, not for object creation syntax.

**Comment by jcouv** [comment]
The fix should ship in 15.6 preview 3.
